WalletWasabi/Blockchain/Transactions/TransactionFactory.cs

```diff
@@ -356,6 +359,24 @@ private void UpdatePSBTInfo(PSBT psbt, SmartCoin[] spentCoins, HdPubKey changeHd
 					psbt.AddKeyPath(changeHdPubKey.PubKey, rootKeyPath, changeHdPubKey.P2wpkhScript);
 				}
 			}
+
+			if (Store is null)
+			{
+				return;
+			}
+
+			foreach (var input in spentCoins)
+			{
+				var coinInputTxID = input.TransactionId;
+				if (Store.TransactionStore.TryGetTransaction(coinInputTxID, out var txn))
+				{
+					var psbtInput = psbt.Inputs.FirstOrDefault(x => x.PrevOut.Hash == coinInputTxID);
+					if (psbtInput != null)
+					{
+						psbtInput.NonWitnessUtxo = txn.Transaction;
+					}
+				}
```

Comment:
What if 2 input comes from one transaction?

Reply:
It will be duplicated on both inputs, sadly...

Reply:
Yes, this is the tx of the current input.

Reply:
if the multiple inputs from the same tx then add the tx to everywhere